include/pybind11/pybind11.h

```diff
@@ -1144,7 +1151,7 @@ class class_ : public detail::generic_type {
     class_ &def(const char *name_, Func&& f, const Extra&... extra) {
         cpp_function cf(method_adaptor<type>(std::forward<Func>(f)), name(name_), is_method(*this),
                         sibling(getattr(*this, name_, none())), extra...);
-        attr(cf.name()) = cf;
```

Comment:
Does anyone know why this line was not `attr(name_) = cf;` ? Just curious.

Reply:
It used to be `attr(name_) = cpp_function( ... )` until [this commit](https://github.com/pybind/pybind11/commit/570822102c037600781da816d1a083db031c8cde#diff-6645501a414a69e6b16abfe4bee9025fR676).

Reply:
Indeed, it's intended change. Had to do `blame` myself. Thanks!